### DISCUSSION TITLE
Is createGetInitialProps needed for a NextJS SPA?

### DISCUSSION THREAD

**Discussion by danielnitu** [mantinedev/mantine#4046] [Q&A] [answered]
Hi, I'm just getting started with Mantine and wanted to understand if it's necessary to use `createGetInitialProps` if my NextJS app will only run on the client and will be served from an S3 bucket or similar.

So I'll never call `getInitialProps` in other places in my code which is why I'm wondering if the code in `pages/_document.tsx` needs it.

**Comment by rtivital** [comment, marked as answer]
You can remove it if you do not need it
